niworkflows/interfaces/bids.py

```diff
@@ -392,110 +401,114 @@ class DerivativesDataSink(SimpleInterface):
-            ext += '.gz'
-        elif self.inputs.compress is False and ext.endswith('.gz'):
-            ext = ext[:-3]
-
-        m = BIDS_NAME.search(src_fname)
-
-        mod = Path(self.inputs.source_file).parent.name
-
+        # Ready the output folder
         base_directory = runtime.cwd
         if isdefined(self.inputs.base_directory):
             base_directory = self.inputs.base_directory
+        base_directory = Path(base_directory).absolute()
+        out_path = base_directory / self.out_path_base
+        out_path.mkdir(exist_ok=True, parents=True)
 
-        base_directory = Path(base_directory).resolve()
-        out_path = base_directory / self.out_path_base / \
-            '{subject_id}'.format(**m.groupdict())
-
-        if m.groupdict().get('session_id') is not None:
-            out_path = out_path / '{session_id}'.format(**m.groupdict())
+        # Ensure we have a list
+        in_file = listify(self.inputs.in_file)
 
-        out_path = out_path / '{}'.format(mod)
-        out_path.mkdir(exist_ok=True, parents=True)
-        base_fname = str(out_path / src_fname)
+        # Read in the dictionary of metadata
+        if isdefined(self.inputs.meta_dict):
+            meta = self.inputs.meta_dict
+            # inputs passed in construction take priority
+            meta.update(self._metadata)
+            self._metadata = meta
 
-        allowed_entities = {}
+        # Initialize entities with those from the source file.
+        out_entities = parse_file_entities(self.inputs.source_file)
+        for drop_entity in listify(self.inputs.dismiss_entities or []):
+            out_entities.pop(drop_entity, None)
+
+        # Override extension with that of the input file(s)
+        out_entities["extension"] = [
+            "".join(Path(orig_file).suffixes).lstrip(".") for orig_file in in_file
+        ]
+
+        if isdefined(self.inputs.compress):
+            compress = listify(self.inputs.compress)
+            if len(compress) == 1 and len(in_file) > 1:
+                compress = compress * len(in_file)
+            out_entities["extension"] = [
+                f"{ext.rstrip('.gz')}" if gz is False
+                else f"{ext.rstrip('.gz')}.gz" if gz is True
+                else ext
+                for ext, gz in zip(out_entities["extension"], compress)
+            ]
+
+        # Override entities with those set as inputs
         for key in self._allowed_entities:
             value = getattr(self.inputs, key)
             if value is not None and isdefined(value):
-                allowed_entities[key] = '_%s-%s' % (key, value)
+                out_entities[key] = value
 
-        formatbase = '{bname}{space}{desc}' + ''.join(
-            [allowed_entities.get(s, '') for s in self._allowed_entities])
+        # Clean up native resolution with space
+        if out_entities.get("resolution", "") == "native" and out_entities.get("space"):
+            out_entities.pop("resolution", None)
```

Comment:
Seems like this is not working. I'll keep an eye on this one ([this build will yell at us](https://app.circleci.com/pipelines/github/oesteban/fmriprep/98/workflows/63e28ac0-b293-4c65-923d-1603a6411773/jobs/5463)).

Anyway, I'm not super-duper with this particular simplification.

Reply:
okay, found the reason. we were stuffing the resolution in the space input, therefore resolution was not defined.

addressed in https://github.com/nipreps/niworkflows/pull/507/commits/401e65093016093e515b40264f3236090855e063, alongside the explicit handling of cohorts.